test/extended/gssapi.sh

```diff
@@ -0,0 +1,205 @@
+#!/usr/bin/env bash
+#
+# Extended tests for logging in using GSSAPI
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+STARTTIME="$(date +%s)"
+
+PROJECT='gssapiproxy'
+TEST_NAME="test-extended/${PROJECT}"
+
+OS_ROOT="$(dirname "${BASH_SOURCE}")/../.."
+cd "${OS_ROOT}"
+source hack/lib/init.sh
+
+os::build::setup_env
+
+os::util::environment::setup_time_vars
+os::util::environment::setup_all_server_vars "${TEST_NAME}"
+os::util::environment::use_sudo
+
+os::log::stacktrace::install
+os::log::start_system_logger
+
+ensure_iptables_or_die ## TODO Is this needed?
+
+reset_tmp_dir
+
+# Allow setting $JUNIT_REPORT to toggle output behavior
+if [[ -n "${JUNIT_REPORT:-}" ]]; then
+    export JUNIT_REPORT_OUTPUT="${LOG_DIR}/raw_test_output.log"
+fi
+
+# Always keep containers' raw output for simplicity
+export JUNIT_GSSAPI_OUTPUT="${LOG_DIR}/raw_test_output_gssapi.log"
+
+os::test::junit::declare_suite_start "${TEST_NAME}"
+
+os::cmd::expect_success 'hack/build-go.sh cmd/oc -tags=gssapi'
+os::cmd::expect_success_and_text 'oc version' 'GSSAPI Kerberos SPNEGO'
+
+function cleanup() {
+    out=$?
+    set +e
+    cleanup_openshift
+
+    # TODO(skuznets): un-hack this nonsense once traps are in a better state
+    if [[ -n "${JUNIT_REPORT_OUTPUT:-}" ]]; then
+      # get the jUnit output file into a workable state in case we crashed in the middle of testing something
+      os::test::junit::reconcile_output
+
+      # check that we didn't mangle jUnit output
+      os::test::junit::check_test_counters
+
+      # use the junitreport tool to generate us a report
+      "${OS_ROOT}/hack/build-go.sh" tools/junitreport
+      junitreport="$(os::build::find-binary junitreport)"
+
+      if [[ -z "${junitreport}" ]]; then
+          echo "It looks as if you don't have a compiled junitreport binary"
+          echo
+          echo "If you are running from a clone of the git repo, please run"
+          echo "'./hack/build-go.sh tools/junitreport'."
+          exit 1
+      fi
+
+      cat "${JUNIT_REPORT_OUTPUT}" "${JUNIT_GSSAPI_OUTPUT}"    \
+        | "${junitreport}" --type oscmd                        \
+                           --suites nested                     \
+                           --roots github.com/openshift/origin \
+                           --output "${ARTIFACT_DIR}/report.xml"
+      cat "${ARTIFACT_DIR}/report.xml" | "${junitreport}" summarize
+    fi
+
+    ENDTIME=$(date +%s); echo "$0 took $(($ENDTIME - $STARTTIME)) seconds"
```

Comment:
i know this was copy-pasted but technically inside of `(())` you don't need `$` on your vars